File: VB/DateNavigator-S92168/MyDateNavigator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Controls

Namespace WindowsApplication22
	Public Class CustomDateNavigator
		Inherits DateNavigator
		Private Class MyDateEditInfoArgs
			Inherits DateEditInfoArgs
			Public Sub New(ByVal calendar As DateEditCalendarBase)
				MyBase.New(calendar)
			End Sub
			Protected Overrides Sub CalcDayNumberCells()
				MyBase.CalcDayNumberCells()
				RemoveInactiveDays()
			End Sub
			Protected Sub RemoveInactiveDays()
				Dim count As Integer = DayCells.Count
				For i As Integer = count - 1 To 0 Step -1
					If (Not DayCells(i).IsActive) Then
						DayCells.RemoveAt(i)
					End If
				Next i
			End Sub
		End Class
		Protected Overrides Function CreateInfoArgs() As DateEditInfoArgs
			Return New MyDateEditInfoArgs(Me)
		End Function
	End Class
End Namespace

File: CS/DateNavigator-S92168/MyDateNavigator.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraScheduler;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Controls;

namespace WindowsApplication22 {
    public class CustomDateNavigator : DateNavigator {
        class MyDateEditInfoArgs : DateEditInfoArgs {
            public MyDateEditInfoArgs(DateEditCalendarBase calendar)
                : base(calendar) {
            }
            protected override void CalcDayNumberCells() {
                base.CalcDayNumberCells();
                RemoveInactiveDays();
            }
            protected void RemoveInactiveDays() {
                int count = DayCells.Count;
                for (int i = count - 1; i >= 0; i--)
                    if (!DayCells[i].IsActive)
                        DayCells.RemoveAt(i);
            }
        }
        protected override DateEditInfoArgs CreateInfoArgs() {
            return new MyDateEditInfoArgs(this);
        }
    }
}
